src/PointwiseFunctions/GeneralRelativity/Python/Bindings.cpp

```diff
@@ -1,21 +1,123 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
+// #include "PointwiseFunctions/GeneralRelativity/Python/Ricci.hpp"
+
 #include <cstddef>
 #include <pybind11/pybind11.h>
 #include <type_traits>
 
 #include "DataStructures/Tensor/TypeAliases.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Lapse.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ProjectionOperators.hpp"
 #include "PointwiseFunctions/GeneralRelativity/Psi4Real.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Ricci.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Shift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalOneForm.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalVector.hpp"
 
 namespace py = pybind11;
 
-namespace GeneralRelativity {
+namespace GeneralRelativity::py_bindings {
+
+namespace {
+template <size_t Dim, IndexType Index>
+void bind_christoffel_impl(py::module& m) {
+  m.def("christoffel_first_kind",
+        static_cast<tnsr::abb<DataVector, Dim, Frame::Inertial, Index> (*)(
+            const tnsr::abb<DataVector, Dim, Frame::Inertial, Index>&)>(
+            &::gr::christoffel_first_kind),
+        py::arg("d_metric"));
+  m.def("christoffel_second_kind",
+        static_cast<tnsr::Abb<DataVector, Dim, Frame::Inertial, Index> (*)(
+            const tnsr::abb<DataVector, Dim, Frame::Inertial, Index>&,
+            const tnsr::AA<DataVector, Dim, Frame::Inertial, Index>&)>(
+            &::gr::christoffel_second_kind),
+        py::arg("d_metric"), py::arg("inverse_metric"));
+}
+void bind_christoffel(py::module& m) {
+  bind_christoffel_impl<1, IndexType::Spatial>(m);
```

Comment:
Do I need to instantiate these functions for multiple Dims, or would Dim 3 be fine since that's the most common use case?

Reply:
I would suggest instantiating on dim=1,2,3, since 1D tests are sometimes done

Reply:
In that case all the other functions in this file should also be instantiated for dim 1,2,3 I think?